### PR TITLE
Add --gpg-sign to `git commit` calls

### PR DESCRIPTION
**Pull request by mattwynne**
This is awkward to test as it integrates with gpg tooling and key database.

### 🤔 What's changed?

* Add `--gpg-sign` to all `git commit` commands (except in tests)
* Add [an indicator](https://git-scm.com/docs/pretty-formats#Documentation/pretty-formats.txt-emGSem) to the `git log` output used in approval tests.
* Added a GPG key for the tests to use (as `user@example.com`), which is installed when the tests run.

### ⚡️ What's your motivation? 

We want all commits to be signed.

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)

### ♻️ Anything particular you want feedback on?

* If you run the tests locally, this will leave a `user@example.com` key in your gpg key list. I can't work how to delete this non-interactively (my `pintentry` fires up even if I use `gpg --delete-secret-keys` in `--batch` mode. Is this OK?
* Maybe there's a way to pass a key to git on the fly instead of cluttering up the global gpg key database? If there is I can't find one.

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by mattwynne** [comment]
of course, the WOMM problem here is that we don't have any GPG keys in the CI environment



need to either not test this feature somehow (make it an option?) or figure out how to set up GPG keys in CI.

**Comment by mpkorstanje** [comment]
Would aliasing the gpg command to use a different home directory while testing work? E.g:

```
alias gpg="gpg --homedir=/c/Users/s156757/.gnupg"
```

From:

https://stackoverflow.com/questions/46863981/how-to-sign-git-commits-from-within-an-ide-like-intellij/46884134#46884134

**Comment by mattwynne** [comment]
Great idea! I'll take a look.

**Comment by mattwynne** [comment]
Sweet, I also discovered https://stackoverflow.com/questions/55576302/how-do-i-pass-homedir-to-git-when-signing-using-gpg which helps to convince git to use the right gpg home directory. Easier that writing shell script wrappers.

File: polyglot-release [hunk 1]
```diff
@@ -225,7 +225,7 @@ changelog release "$NEW_VERSION" \
   --output CHANGELOG.md
 
 if [[ -z $NO_GIT_COMMIT ]]; then
-  git commit --quiet --all --message="Prepare release v$NEW_VERSION"
+  git commit --gpg-sign --quiet --all --message="Prepare release v$NEW_VERSION"
   git tag "v$NEW_VERSION"
```
Comment (mpkorstanje) [inline polyglot-release:229]: What about signing the tag?

Comment (mattwynne) [reply polyglot-release:229]: Good point, yes we should do that.

Comment (mattwynne) [reply polyglot-release]: Resolved in 70f5884fb563c8754aebd73ead43aa4ca00c86d7

**Comment by mattwynne** [comment]
Do you think we should have a pre-flight check for GPG keys? I'm not sure how clear the feedback will be if you don't have them set up.

**Review by mpkorstanje — APPROVED** [review]
Looks good!

**Comment by mpkorstanje** [comment]
> Do you think we should have a pre-flight check for GPG keys?

Would be nice.  We don't have it documented that you should set it up so it's either writing a check or documentation.

**Pull request merged**